keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,958 @@
+	AppliedTo   AppliedTo
+	Ingress     []DefaultNetworkPolicyIngressRule
+	Egress      []DefaultNetworkPolicyEgressRule
+}
+
+type DefaultNetworkPolicyIngress/EgressRule struct {
+	Ports            []networkingv1.NetworkPolicyPort
+	OnlyFrom/OnlyTo  []networkingv1.DefaultNetworkPolicyPeer
+}
+
+type DefaultNetworkPolicyPeer struct {
+	PodSelector  *metav1.LabelSelector
+	// required if a PodSelector is specified
+	Namespaces   *networkingv1.Namespaces
+	IPBlock      *IPBlock
+}
+```
+
+Most structs above are very similar to NetworkPolicy and quite self-explanatory.
+One detail to notice is that in the DefaultNetworkPolicy Ingress/Egress rule spec,
+the peers are created in a field named `OnlyFrom`/`OnlyTo`, as opposed to `To`/`From`
+in ClusterNetworkPolicy. We chose this naming to better hint policy writers about
+the isolation effect of DefaultNetworkPolicy on the workloads it applies to.
+
+### Shared API Design
+The following structs will be added to the `networking.k8s.io` API group and
+shared between `ClusterNetworkPolicy` and `DefaultNetworkPolicy`:
+
+```golang
+type AppliedTo struct {
+	// required if a PodSelector is specified
+	NamespaceSelector   *metav1.LabelSelector
+	// optional
+	PodSelector         *metav1.LabelSelector
+}
+
+type Namespaces struct {
+	// Self and Selector are mutually-exclusive
+	Self       bool
+	Selector   *metav1.LabelSelector
+}
+```
+
+#### AppliedTo
+The `AppliedTo` field replaces `PodSelector` in NetworkPolicy spec, as means to specify
+the target Pods that this cluster-scoped policy (either `ClusterNetworkPolicy` or
+`DefaultNetworkPolicy`) applies to.
+Since the policy is cluster-scoped, the `NamespaceSelector` field is required.
+An empty `NamespaceSelector` (namespaceSelector: {}) selects all Namespaces in the Cluster.
+
+#### Namespaces
+The `Namespaces` field replaces `NamespaceSelector` in NetworkPolicyPeer, as
+means to specify the Namespaces of ingress/egress peers for cluster-scoped policies.
+For selecting Pods from specific Namespaces, the `Selector` field under `Namespaces`
+works exactly as `NamespaceSelector`. The `Self` field is added to satisfy the
+specific needs for cluster-scoped policies:
+
+__Self:__ An optional field, which evaluates to false by default.
+When `self: true` is set, no selectors can be present concurrently. This is a
+special keyword to indicate that the rule only applies to the Namespace for
+which the ingress/egress rule is currently being evaluated upon. Since the Pods
+selected by the ClusterNetworkPolicy appliedTo could be from multiple Namespaces,
+the scope of ingress/egress rules whose `namespace.self=true` will be the Pod's
+own Namespace for each selected Pod.
+Consider the following example:
+
+- Pods [a1, b1] exist in Namespace x, which has labels `app=a` and `app=b` respectively.
+- Pods [a2, b2] exist in Namespace y, which also has labels `app=a` and `app=b` respectively.
+
+```yaml
+apiVersion: networking.k8s.io/v1alpha1
```

Comment:
I don't have enough knowledge about this, @liggitt is it possible to enable an api group version that was already deprecated, in this case `networking.k8s.io/v1alpha1` ?